src/nvim/os/env.c

```diff
@@ -127,11 +131,19 @@ void os_get_hostname(char *hostname, size_t len)
     strncpy(hostname, vutsname.nodename, len - 1);
     hostname[len - 1] = '\0';
   }
-#else
-  // TODO(unknown): Implement this for windows.
-  // See the implementation used in vim:
-  // https://code.google.com/p/vim/source/browse/src/os_win32.c?r=6b69d8dde19e32909f4ee3a6337e6a2ecfbb6f72#2899
-  *hostname = '\0';
+#endif
+#ifdef WIN32
+  WCHAR hostname_unicode[MAX_COMPUTERNAME_LENGTH + 1];
+  WCHAR hostname_length = (WCHAR)len;
+  if (GetComputerName(hostname_unicode, &hostname_length) == 0) {
+    *hostname = '\0';
+  } else {
+    int conversion_result = utf16_to_utf8(hostname_unicode,
+                                          &hostname);
```

Comment:
This doesn't work because `hostname` is reassigned by `utf16_to_utf8`, so the original pointer passed to `os_get_hostname()` isn't updated. Instead a temporary pointer should be passed to `utf16_to_utf8` then copied to `hostname`.